mypy/myunit/suite.py

```diff
@@ -0,0 +1,79 @@
+from typing import Callable, List, Tuple, Union
+
+import os
+import tempfile
+
+from mypy.myunit.errors import SkipTestCaseException
+
+
+class TestCase:
+    def __init__(self, name: str, suite: 'Suite' = None,
+                 func: Callable[[], None] = None) -> None:
+        self.func = func
+        self.name = name
+        self.suite = suite
+        self.old_cwd = None  # type: str
+        self.tmpdir = None  # type: tempfile.TemporaryDirectory
+
+    def run(self) -> None:
+        if self.func:
+            self.func()
+
+    def set_up(self) -> None:
+        self.old_cwd = os.getcwd()
+        self.tmpdir = tempfile.TemporaryDirectory(prefix='mypy-test-',
+                dir=os.path.abspath('tmp-test-dirs'))
+        os.chdir(self.tmpdir.name)
+        os.mkdir('tmp')
+        if self.suite:
+            self.suite.set_up()
+
+    def tear_down(self) -> None:
+        if self.suite:
+            self.suite.tear_down()
+        os.chdir(self.old_cwd)
+        self.tmpdir.cleanup()
+        self.old_cwd = None
+        self.tmpdir = None
+
+
+TestUnion = Union[TestCase, Tuple[str, 'Suite']]
```

Comment:
Can you come up with a more descriptive name for this? Not sure what would work, but maybe something like `Testable` (not sure really).